src/System.Device.Gpio/System/Device/Gpio/GpioController.cs

```diff
@@ -12,7 +12,7 @@ namespace System.Device.Gpio
     /// <summary>
     /// Represents a general-purpose I/O (GPIO) controller.
     /// </summary>
-    public sealed class GpioController : IDisposable
+    public class GpioController : IDisposable
```

Comment:
I don't think that this class can satisfy the scenario that I am using in the pi-top libraries. I need a controller that is just a proxy to a real controller. With this implementation anything conforming with the controller interface will have to call base class constructors that will try to create a drive. This part of the class contracts seems a little bit too restrictive.

Reply:
can you pass null driver and override all methods?

Reply:
sounds little bit aggressive if null is allowed then then internal calls should be something like `bool TryOpenPin()`, that should wrap the driver access. Such protected methods will then potentially address the previous questions about the driver.

I like your suggestion, if we can then protected the access to the driver api via a Try* set of methods that handles the null driver that would be an easier pattern for implementors to adopt.

Reply:
@colombod Can you elaborate a bit more on your use case?

It is possible to pass null as the driver, or a DummyDriver() [will be provided in one of my next PRs]. All methods accessing the `_driver` member are virtual, so if you override all of them (without calling the base implementation, of course), you should be fine.

Reply:
in the pi-top libraries we use a gpio controller factory, this returns a wrapper that tracks it's onw pin open/close logic and also events subscriptions. All work si delegated to the concrete controller.
on dispose such controller "projections" dispose all things that where opened via the proxy. That controller will require to have no driver and track the pins open requests. If passing null is safe in this PR then is good for me. And will be able to adopt this new class and finally get rid of the interfaces we introduced. This is the [shared controller class](https://github.com/pi-top/pi-top-4-.NET-SDK/blob/master/src/PiTop.Abstractions/SharedGpioController.cs)

Reply:
Yea, I think this should all fit now. You do not need any of your intermediate classes. Even `SharedGpioController` doesn't seem to do anything more than what is aleady implemented.